cmip6_downscaling/methods/gard/tasks.py

```diff
@@ -0,0 +1,281 @@
+    """
+    ds_hash = str_to_hash(
+        str(xtrain_path)
+        + str(ytrain_path)
+        + str(xpred_path)
+        + str(scrf_path)
+        + run_parameters.run_id_hash
+        + str(dim)
+    )
+    target = intermediate_dir / 'gard_fit_and_predict' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        return target
+
+    # load in datasets
+    xtrain = xr.open_zarr(xtrain_path)
+    ytrain = xr.open_zarr(ytrain_path)
+    xpred = xr.open_zarr(xpred_path)
+    scrf = xr.open_zarr(scrf_path)
+    # make sure you have the variables you need in obs
+    for v in xpred.data_vars:
+        assert v in ytrain.data_vars
+    # data transformation (this wants full-time chunking)
+    # transformed_obs is for the training period
+
+    # # we need only the prediction GCM (xpred), but we'll transform it into the space of the
+    # # transformed interpolated obs (xtrain)
+    # # Create a template dataset for map blocks
+    # # This feals a bit fragile.
+    template_var = list(xpred.data_vars.keys())[0]
+    template_da = xpred[template_var]
+    template = xr.Dataset()
+    for var in [run_parameters.variable]:
+        template[var] = template_da
+
+    out = xr.map_blocks(
+        _fit_and_predict_wrapper,
+        xtrain,
+        args=(ytrain, xpred.rename({'time': 't2'}), scrf.rename({'time': 't2'}), run_parameters),
+        kwargs={'dim': dim},
+        template=template,
+    )
+    out.attrs.update({'title': 'gard_fit_and_predict'}, **get_cf_global_attrs(version=version))
+    out = dask.optimize(out)[0]
+    t = out.to_zarr(target, compute=False, mode='w', consolidated=False)
+    t.compute(retries=5)
+    zarr.consolidate_metadata(target)
+    return target
+
+
+@task(log_stdout=True)
+def read_scrf(prediction_path: UPath, run_parameters: RunParameters):
+    """
+    Read spatial-temporally correlated random fields on file and subset into the correct spatial/temporal domain according to model_output.
+    The random fields are stored in decade (10 year) long time series for the global domain and pre-generated using `scrf.ipynb`.
+
+    Parameters
+    ----------
+    prediction_path : UPath
+        Path to prediction dataset
+    run_parameters : RunParameters
+
+
+    Returns
+    -------
+    scrf : xr.DataArray
+        Spatio-temporally correlated random fields (SCRF)
+    """
+    # TODO: this is a temporary creation of random fields. ultimately we probably want to have
+    # ~150 years of random fields, but this is fine.
+
+    ds_hash = str_to_hash(
+        "{obs}_{variable}_{latmin}_{latmax}_{lonmin}_{lonmax}_{predict_dates[0]}_{predict_dates[1]}".format(
+            **asdict(run_parameters)
+        )
+    )
+
+    target = intermediate_dir / 'scrf' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        return target
+    prediction_ds = xr.open_zarr(prediction_path)
```

Comment:
@jhamman - do we have two scrf for tmin and tmax? do we need some logic here so that when `tmin` comes up it knows to find it? @norlandrhagen this is for future reference, not needed now.

Reply:
we should be all set for this. We've already staged scrfs for tasmin, tasmax, and pr.